### ISSUE TITLE
Gradient Penalty is very high in the start

### ISSUE THREAD

**Issue by inspirit** [lucidrains/gigagan-pytorch#46]
Hi!

i was running few experiments and noticed that GP is extremely hight in first few 100 steps.
GP > 60000, and then gradually going down to around GP = 20

is it normal behaviour? In my previous experience with StyleGan GP was small in the beginning

**Comment by inspirit** [comment]
hey! i closed it because the results were fine, but i did not train it for long, i'm borrowing different ideas from various GAN implementations for some experiments. GigaGan discriminator is definitely not the most stable and D/MSD/GP are fluctuating a lot

**Comment by lucidrains** [comment]
@inspirit yes, i think it is the multi-scale causing it, last i was training on my toy datasets. if you turn it off by setting the loss weight to 0, the training resembles stylegan a lot (could be remembering wrong)

**Comment by lucidrains** [comment]
@inspirit i think it is necessary to achieve the paper's results

**Comment by lucidrains** [comment]
@inspirit are you a phd student btw? we run into each other a lot

**Comment by inspirit** [comment]
Haha I'm long time past that age i believe :) Just freelance researcher and technical adviser for CV/ML stuff

**Comment by lucidrains** [comment]
@inspirit ah got it :) well, glad to have you reviewing my code from time to time!

**Comment by inspirit** [comment]
> @inspirit i think it is necessary to achieve the paper's results

in my opinion adaptive mod-conv is more useful and can be fitted in lots of ideas...

**Comment by lucidrains** [comment]
@inspirit yea i think both may be important
